javascript/prototypingCharacterClasses.md

```diff
@@ -0,0 +1,56 @@
+#Prototype Your (Character) Class#
+
+Continuing in using Javascript to build D&D characters, I decided to use prototypes.
+
+```javascript
+function Character (name, dr, profession, hd, castingAbility) {
+	this.name = name;
+	this.profession = profession;
+	this.hd = hd;
+	this.dr = dr;
+	this.lvl = 1;
+	this.profBonus = 2;
+	this.castingAbility = castingAbility;
+	this.exp = 0;
+	this.abilities = new Array(6);
+	this.mod = this.abilities.map(function (score){ return Math.floor((score / 2) - 5); } );
+	this.ac = 10 + this.mod[1];
+	this.hp = this.hd + this.mod[2] + this.dr;
+	this.spellSave = 8 + this.profBonus + this.mod[castingAbility];
+	this.spellAttack = this.profBonus + this.mod[castingAbility];
+}
+``` 
+	
+In using a prototype over an object we gain the ability to efficiently store identical functions common to all characters without bloating objects.
+
+```javascript
+Character.prototype = {
+	level: function (){ 
```

Comment:
Thoughts on renaming this function `levelUp` to make it obvious this is changing the character, not just retrieving the level?